### DISCUSSION TITLE
함수에 객체를 전달할 때 참조에 의한 전달과 값에 의한 전달의 차이를 설명해보세요.

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#66] [원시 값과 객체의 비교]
- 함수에 객체를 전달할 때 참조에 의한 전달과 값에 의한 전달의 차이를 설명해보세요.

**Comment by pipisebastian** [comment]
> 함수에 객체를 전달할 때 참조에 의한 전달과 값에 의한 전달의 차이를 설명해보세요.

- 값에 의한 전달은, 원시값을 복사하여 함수의 인자로 전달합니다. 그렇기에 함수내부에서 값을 변경하더라도, 원본값에는 변화가 없습니다.
- 참조에 의한 전달은, 메모리 주소가 전달됩니다. 이에 값을 변경하면 원본값까지 수정되는 문제가 있습니다.

**Comment by publdaze** [comment]
> - 함수에 객체를 전달할 때 참조에 의한 전달과 값에 의한 전달의 차이를 설명해보세요.
- 함수에 객체를 전달할 때는 참조에 의한 전달이되기 때문에 함수 내부에서 프로퍼티 값이 변경될 수 있음
- 원시 값의 경우는 값에 의한 전달이 이루어지기 때문에 변경을 하더라고 전달한 인자 자체의 값은 변경되지 않음
